crdb/error.go

```diff
@@ -0,0 +1,60 @@
+package crdb
+
+import "fmt"
+
+type txError struct {
+	cause error
+}
+
+// Error implements the error interface
+func (e *txError) Error() string { return e.cause.Error() }
+
+// Cause returns the error encountered by the "ROLLBACK TO SAVEPOINT
+// cockroach_restart" statement. This method also implements the internal
+// pkg/errors.causer interface, so TxnRestartError works with pkg/errors.Cause().
+func (e *txError) Cause() error { return e.cause }
+
+// AmbiguousCommitError represents an error that left a transaction in an
+// ambiguous state: unclear if it committed or not.
+type AmbiguousCommitError struct {
+	txError
+}
+
+func newAmbiguousCommitError(err error) *AmbiguousCommitError {
+	if err == nil {
```

Comment:
all these nil checks are "more general" than necessary - it shouldn't be valid to pass nil values in any of these parameters. I'd just remove them all, but @andreimatei should weigh in.